### ISSUE TITLE
Table view for data browser

### ISSUE THREAD

**Issue by lesslogic** [pudeIko/piva#50]
**Table view** for rapidly comparing and sorting data files with the following features:
- **Basic spreadsheet operations:**
  - Customizable column ordering and visibility
  - Sort rows by column value (ascending descending)
  - Filter results by column values of one or more columns
- **Ability to export the table in various formats** (.csv, .xlsx, .html, .md...)
- **Multi-selection of files** to perform simultaneous operations on (namely, opening all selected files in viewers with one click).

I think it will work best if the user browses the directory hierarchy and specifically chooses which files/folders to import to the table. For one thing, this would be the simplest way to let a user sort/compare data located in different directories.

My strong personal preference would be to have the table **display "lab-specific" metadata, as opposed to PIVA-standardized metadata**. Reasoning: 
- The browser becomes much more powerful if we don't make assumptions about *what* metadata the users are interested in. Example: Fluence or delay time in pump-probe experiments.
- I expect that the most common use case is to browse files from a single lab/instrument at a time, in which case, there is no problem with conflicting column labels. Even if such cases arise, I think there can be reasonable solutions or workarounds.
- Conversely, I don't see an advantage to showing PIVA-style metadata at the level of the browser.
- In summary, the goal should be to help people sort through data to find what's interesting to them. The metadata that users record reflects the information that is valuable to them, in the terminology they're used to. We shouldn't hinder them by changing/constraining that metadata in the browser.

I don't have a strong opinion about what should happen with the directory tree browsing panel. It could stay on the left side, become a "dockable" component that the user can move around, or become a dialog that only appears when opened from the main data browser panel.
